src/include/miopen/miopen_internal.h

```diff
@@ -104,6 +106,23 @@ MIOPEN_EXPORT miopenStatus_t miopenGetConvolutionFindMode(
 
 /* End of Find Mode API */
 
+/*! @brief Get extra workspace for CK's backward weight kernel.
+ *
+ * @param miopenAlphaBetaCase_t type of alpha beta case
+ * @param  miopenDataType_t data type
+ * @param  size_t input channel
+ * @param  size_t output channel
+ * @param  size_t output tensor length
+ * @param  size_t buffer_size for CK Backward weights work space
+ */
+extern "C" miopenStatus_t
+miopenConvolutionCKBackwardWeightsGetWorkSpaceSize(const miopenAlphaBetaCase_t ab_case,
```

Comment:
@bghimireamd @amberhassaan @junliume Excuse me for late asking, but what is the purpose of this function? The question seems important to me.

Reply:
@atamazov : this came up when supporting `alpha/beta` scaling parameters for convolution recently. All of our current `GetWorkspaceSize` functions do not take `alpha` or `beta` but we ran into a situation where the nature of these parameters affected the workspace size calculation, so we added this new function. 

@bghimireamd : this should probably be named `miopenConvolutionAB...`. CK is an implementation detail that users don't care about.

Reply:
@amberhassaan @bghimireamd Aha, this `miopenConvolutionCKBackwardWeightsGetWorkSpaceSize()` is only for internal use in our library, right? If this is so, then I do not see any reason to declare this functionality in the "internal C API" header. This file contains some APIs that we created but then decided not to open to end users for a while (I don't remember exactly why). In any case, miopen_internal.h contains candidates for miopen.h. Can we implement it as a C++ function, as usual? That would simplify the code, I guess.

Reply:
It is also intended as a `beta API` for external users when they are using scaling parameters. May be we should put it in `miopen.h` under the `beta API` section?

Reply:
@atamazov should I move it to `miopen.h` under `beta API` or just move it as function inside `impilicitgemm_ck_util.hpp`?

Reply:
@amberhassaan @bghimireamd Unfortunately, I am unable to find the description of the key design choices in this PR. Therefore I had to review this PR in some more detail to answer the question.

The short answer: No. We need to introduce some other thing into our public API.

Longer answer: The problem is that A/B (alpha/beta) is currently unknown to `miopenFindConvolution*`, so it works as if A/B have default values. This violates the guarantees we must provide to the subsequent `miopenConvolution*`, so it won't work as expected if non-default A/B passed. This problem cannot be resolved with things like miopenConvolutionCKBackwardWeightsGetWorkSpaceSize().

AFAICS the correct way to implement this feature (for Find 1.0 API) includes updating the public API. We have to add possibility to store "a sign of non-default A/B" (something like `ab_case` from this PR) in the ConvolutionDescriptor. This way "a sign of non-default A/B" becomes known to `miopenFindConvolution*` and `miopenGetWorkspaceSize*`. This opens possibility to produce correct Invokers suring Find phase, and also makes additional API calls like "get additional workspace size" unnecessary.

[Notice] FYI current ImmediateMode API does not allow passing alpha/beta.

/cc @junliume @JehandadKhan @CAHEK7